sdk/java/pulumi/src/main/java/com/pulumi/resources/internal/Stack.java

```diff
@@ -0,0 +1,109 @@
+package com.pulumi.resources.internal;
+
+import com.pulumi.core.Output;
+import com.pulumi.core.internal.annotations.InternalUse;
+import com.pulumi.deployment.internal.DeploymentInternal;
+import com.pulumi.resources.ComponentResource;
+import com.pulumi.resources.ComponentResourceOptions;
+import com.pulumi.resources.ResourceTransformation;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+@InternalUse
+@ParametersAreNonnullByDefault
+public final class Stack extends ComponentResource {
+
+    /**
+     * The type name that should be used to construct the root component in the tree of Pulumi resources
+     * allocated by a deployment. This must be kept up to date with
+     * "github.com/pulumi/pulumi/sdk/v3/go/common/resource/stack.RootStackType".
+     */
+    @InternalUse
+    public static final String RootPulumiStackTypeName = "pulumi:pulumi:Stack";
+
+    /**
+     * The outputs of this stack, if the <code>callback</code> callback exited normally.
+     */
+    private final Output<Map<String, Output<?>>> outputs;
+
+    /**
+     * Create a Stack with stack outputs defined in derived class constructor.
```

Comment:
The comment is weird, what is "derived class constructor"? Stack is now final right?

Reply:
fixed